tests/core/uc20-recovery/task.yaml

```diff
@@ -0,0 +1,85 @@
+summary: verify booting into recovery mode on UC20 systems
+
+systems: [ ubuntu-core-20-* ]
+
+restore: |
+  rm -f /writable/systems.json.run
+  rm -f /writable/systems.label
+  rm -f system-info systems.json
+
+  snap remove http jq
+
+debug: |
+  if [ -e systems.json ]; then
+      cat systems.json
+  fi
+
+execute: |
+  if [[ "$SPREAD_REBOOT" == "0" ]]; then
+      echo "In run mode"
+
+      snap install jq
+      # devmode as the snap does not have snapd-control
+      snap install test-snapd-curl --devmode --edge
+
+      MATCH 'snapd_recovery_mode=run' < /proc/cmdline
+      # verify we are in run mode via the API
+      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/system-info > system-info
+      jq -r '.result["system-mode"]' < system-info | MATCH 'run'
+
+      echo "Obtain available systems"
+      test-snapd-curl.curl -s --unix-socket /run/snapd.socket http://localhost/v2/systems > systems.json
+      # TODO:UC20: there is only one system for now
+      jq .result.systems[0].current < systems.json | MATCH 'true'
+      label="$(jq -r .result.systems[0].label < systems.json)"
+      test -n "$label"
+      test "$label" = "$(date +%Y%m%d)"
```

Comment:
mmm this seems a bit racy, i.e. since it takes a few minutes to create the image, what if the image was created at 23:59 and so the date is may 12th, but by the time we hit this test, the date is now 00:01 may 13th? I think a better check here would be that the label matches `/var/lib/snapd/seed/systems/<number>`